### DISCUSSION TITLE
Frigate+ Base Model 2024.2 Update

### DISCUSSION THREAD

**Discussion by blakeblackshear** [blakeblackshear/frigate#13023] [General]
The base model in Frigate+ has now been updated to 2024.2. All model requests going forward will use this version. In order to get a new model, you will need to submit a model request.

The 2024.2 base model update was focused on reducing known false positives by selecting the highest scoring false positives that were reproducible with the 2024.1 base model. To date, nearly 500,000 false positive examples have been submitted by users. The previous base model reproduces 14% of these false positives and the new base model reduces that to 7%. One of the most common false positives was a person being identified as a dog while bent over, so hopefully that should be less common going forward.

I want to reiterate that you should not report false positives for objects that are in locations you want to ignore. You aren't likely to be able to convince your model that a person outside of your zones is not a person at all. This also makes it difficult for me to find good false positive examples to improve the base model.

I wanted to expand the label set with this release, and I have spent a significant amount of time adding labels to the existing base model training set already. However, I underestimated how significant the impact of adding new labels would be on the entire workflow. I am now working on the following updates which I believe are necessary for the an expanded label set:

- AI suggested labels
- Fixed suggestions for objects consistently in the same locations
- Label/camera specific review workflows
- Zooming
- Labels available only for annotation and not in models yet

Once these are done, adding additional labels will be much more straightforward.

In addition, I have almost finished refactoring the annotator in Frigate+ into a separate open source component library so it can be integrated directly into Frigate.

**Comment by jakeycrx** [comment]
When are you planning on implementing zooming?

**Reply by blakeblackshear** [reply]
I already have it working as a part of moving the annotator out into its own component library, so once I get that integrated back into the frontend.

**Reply by jakeycrx** [reply]
Good stuff. Will we have the ability to scroll through images with the new frontend?

**Reply by blakeblackshear** [reply]
What I have in mind should cover the use cases I have heard so far and what's needed to go back and annotate with new labels.

**Comment by autumnwalker** [comment]
This is great news, thanks Blake!

What does "Fixed suggestions for objects consistently in the same locations" mean? I'm making assumptions, but wanted to clarify.

**Reply by NickM-27** [reply]
It means objects that are in the same location like a car that is often parked in the same spot can be a suggested box automatically

**Reply by tube0013** [reply]
yes please, I often avoid submitting images from my front cameras due to all the cars visible and not wanting to annotate each one every time.

**Reply by autumnwalker** [reply]
This is what I was assuming / hoping for! I'm very excited for this feature!

**Reply by bagobones** [reply]
> yes please, I often avoid submitting images from my front cameras due to all the cars visible and not wanting to annotate each one every time.

I have a DUO2 and live on a very straight section of street with houses all the way down as far as the camera can see.. I have SO many cars in view at all times of day..

**Reply by jakeycrx** [reply]
I mask my images before submitting to frigate+ with no adverse effects

**Comment by alexlmiller** [comment]
Thanks for the update Blake - how often do you plan to do the base model updates going forward? Just trying to plan out how to use my trainings

**Reply by NickM-27** [reply]
I believe it says in the docs, but for a while now it has been said that there will be quarterly updates

**Comment by 18923** [comment]
From the Frigate + doc. . .

**INFO
The baseline model isn't directly available after subscribing. This may change in the future, but for now you will need to submit a model.** 

Is this likely to change anytime soon? I’d be willing to pay for a base model.

**Comment by adamburgoyne** [comment]
2024.1 has been great for me, I'd originally bought into Frigate+ in hope of reducing false positives due to cobwebs on the cameras and our shady looking BBQ cover in the corner of the garden. I can't even remember the last time I had a false positive now, excited to give 2024.2 a go!

**Reply by ElectronicBattle** [reply]
I've the same problem (esp. at this time of year, worsening into Autumn!) and, given a few example frames below which are from an untrained Frigate mistakenly identifying "person" in each shot, do you think a _trained_ model from 2024.1/.2 would improve things similarly well?
![image](https://github.com/user-attachments/assets/99c03850-58eb-4a3f-9564-34e923dafc48)
![image](https://github.com/user-attachments/assets/d5e91d5f-3e2d-4bee-9838-ad2d5f7467cc)
![image](https://github.com/user-attachments/assets/1e7463ae-af1b-4c42-811e-a748e47ae09f)
![image](https://github.com/user-attachments/assets/826e1622-4d8d-40a2-8ffa-d7c8345d567f)

**Reply by bagobones** [reply]
Turning off the IR on the camera and getting a seperate IR spot light might help more so the camera view isn't constantly blocked and needing a cleaning.

**Comment by F1VWdk** [comment]
How to report false positives with the new UI?

**Reply by NickM-27** [reply]
you use the Frigate+ page in the sidebar

**Reply by F1VWdk** [reply]
Ohh, great! Sorry I missed that. I did not have snapshots enabled so there was no images in the F+ sidebar page

**Comment by Maudfer** [comment]
Niiice! Thanks for all your effort!

This is amazing timing! I've been procrastinating getting into Frigate+ stuff for half a year, and then this weekend something got into me and I've submitted 3000 verified images across 2 model trainings. The improvement is impressive.

I imagine I may be a bit of an edge case because I put cameras on the ceiling of every room in my house and am trying to get Frigate to be a presence sensor of sorts. There are some unique challenges to it, like identifying people sleeping under blankets haha.

It's already amazing! It's starting to be able to tell me apart from my dog even when I'm, say, hugging her on the sofa.

**Comment by jackez** [comment]
What about OpenVino support?

**Reply by blakeblackshear** [reply]
It's still in the works. I upgraded the version of OpenVino in 0.14 to the latest version to prepare for Frigate+ support.

**Reply by jackez** [reply]
Thanks! Is it now delayed until the next base model update, which is expected in about 3 months? Or could it happen earlier?"

I also considering give it a shot myself. E.g. I found [this](https://docs.openvino.ai/2022.3/openvino_docs_MO_DG_prepare_model_convert_model_Convert_Model_From_TensorFlow.html#converting-a-tensorflow-model) official guide on TensorFlow -> OpenVINO IR conversion. And [this](https://medium.com/openvino-toolkit/importing-pytorch-and-tensorflow-models-into-openvino-376e2336324c) article stating that TensorFlow model could be loaded directly (converted on the fly). Do you think those ways might work or it's more complicated than that?

**Reply by ivanjx** [reply]
@jackez i think frigate uses `.bin` and `.xml` for ssdlite v2 and not `.pb` (tensorflow) directly.

anyways, as soon as i can use the F+ model with openvino CPU/GPU, i will subscribe.

**Reply by ivanjx** [reply]
just tried openvino frigate+ model and it is amazing compared with the generic model. though it uses yolonas. working fine with my i7 7700T.

**Reply by tpjanssen** [reply]
@blakeblackshear is an openvino model generated nowadays when you request for a new model?

**Reply by NickM-27** [reply]
when you request a model you specify if you want mobiledet or yolonas format

**Reply by tpjanssen** [reply]
OK. This is added recently?

**Reply by NickM-27** [reply]
this past weekend I believe

**Comment by MStcz** [comment]
From Frigate+ documentation:
> **Available label types**
Frigate+ models support a more relevant set of objects for security cameras. Currently, only the following objects are supported: person, face, car, license_plate, amazon, ups, fedex, package, dog, cat, deer. Other object types available in the default Frigate model are not available. Additional object types will be added in future releases.

A motorcycle is not available yet. Does it make sense to label a motorcycle as a car or a person to be able to detect them by a custom model or it is necessary to wait for future release with more object types?

**Reply by Maudfer** [reply]
I always label the person on top of the motorcycle (keeping the selection box around the rider only, not the whole vehicle). It's gotten nearly perfect at detecting that on my model. When we have motorcycle labels I'll probably select only the vehicle, so that it also detects parked ones.

I'm sure that's not the only way to go about, but it makes sense in my mind.

**Comment by YBonline** [comment]
Just figured I'd give some feedback... Night one with 2024.2, I've had 0 false positives for the first time ever. It is a little less sensitive to the true positives as well though, but the misses are generally what I would consider "hard" (far from being in the primary area of the frame, in the background in a distance).

**Reply by Maudfer** [reply]
My experience is very similar! Noticeable decrease in false positives. Lost some of the most impressive hard to get true positives, but they are edge case enough for me to think it was well worth it!

**Comment by spridgeon84** [comment]
Really excited about the AI suggested label feature. Great work!

Would also love to see the interface within the self-hosted portion of frigate automatically advance to the next picture when confirming/rejecting object detections, as well as keyboard shortcuts!

**Comment by ne0ark** [comment]
Can you please add this new to the frigate+ website? It makes it hard to hunt these upgrades.

**Comment by petervk** [comment]
How are things progressing on this front? Will any of the prerequisites for additional labels be available in the next point release?

**Reply by blakeblackshear** [reply]
I am almost done making the backend changes to track verification of individual labels on each image instead of a single verified flag on the image. There will be some small related UI changes when that goes live. There will no longer be a simple "verified images" count since verified is set per label. Once that is done, I need to update images view and annotator so you can review a set of images again for a new/existing label. Then I can add a lot of new label options for annotating. At that point, I can start incorporating new labels into the base training set so they are supported in the model.

**Comment by Xyrodileas** [comment]
Hello,
Thanks again for this great release ! Looking forward to the AI suggested labels. Will save me tons of time !

Regarding the new model (2024.2):
I've been testing the new model for a couples of days now, and I've notice a significant difference for true positive, it seems that it's missing a lots of "persons" now walking in the street, where with 2024.1 I would get more false positive, but it would not miss real person.

I've already re-train the model a second time with additional samples that were a missed true positive, but with no improvement.
Let me know if I can provide any additional information.

**Comment by jakeycrx** [comment]
Do you know roughly when the zooming feature will be available? I would like to train a batch of new images, but I'm holding out for the zoom feature first.

**Reply by blakeblackshear** [reply]
I have a few more bugs to fix when drawing a bounding box while zoomed in, but otherwise it's working. I'm hoping to fix that and publish the open source library within a week. Then I can update the Frigate+ UI.

**Reply by blakeblackshear** [reply]
Zooming is now available. Let me know if you have any issues.

**Reply by jakeycrx** [reply]
I have a big queue of images to annotate, so it will be getting a thorough test!

Just to confirm, we don't have to worry about annotating everything now?

![image](https://github.com/user-attachments/assets/b3129c85-1003-4172-a785-ecfceb6b1f4b)

I can zoom in like this and only label the cat?

![image](https://github.com/user-attachments/assets/2eb41bfe-1348-4ba7-9048-3a07db267ecd)

**Reply by jakeycrx** [reply]
Should it save with the zoomed view?

For example, I zoomed in, verified and saved:

![image](https://github.com/user-attachments/assets/880278b5-bd4d-4d40-b4f0-5cc9f2b8d7c1)

But when I reopen the image, it's not zoomed:

![image](https://github.com/user-attachments/assets/4b0e0e5d-ee25-46d5-9721-1a2423ead6fd)

**Reply by blakeblackshear** [reply]
This has nothing to do with avoiding annotations for everything in the image. It's just a way to zoom in on smaller objects. Everything in the image still needs to be annotated.

**Reply by jakeycrx** [reply]
OK. I think you can see my predicament here. I end up labelling cars over and over again when my car scores are in the mid to high 90's and don't require anymore training. I was hoping this would mean I could avoid that.

**Reply by NickM-27** [reply]
As far as I understand objects that you care about will always need to be labeled for training , otherwise the model would be taught that that is not a car for example. What you are looking for is the featured mentioned in the OP called AI suggested labels which will use a model to apply suggestions for objects so you don't have to label all by hand

**Reply by jakeycrx** [reply]
We discussed a cropping or masking feature a little while back. I think that's where my misunderstanding of the zoom feature stemmed from. AI labelling would definitely speed things up a bit.

**Reply by jakeycrx** [reply]
It won't cause any issues if I mask with image editing software then feed that into Frigate+ will it? This is much quicker than annotating 5-6 cars. I've edited the JPG here just for demonstration purposes.

![image](https://github.com/user-attachments/assets/271e7cb8-89df-4142-8bf3-bfa8b68f28ff)

**Reply by blakeblackshear** [reply]
I can't say with 100% certainty since I haven't run an test, but I think this will be fine. There are definitely other users who have privacy masks like this for their cameras. The difference is that their privacy masks are also there at runtime and yours won't. In practice, it shouldn't really be all that different than when the cat is near the edge of the frame. I would definitely suggest using the clean snapshots where you don't have the bounding box drawn on the image.

**Reply by jakeycrx** [reply]
I'll give it a try and see it goes.

I only posted the JPG so you could see the bounding box. I'm adding my masks to the PNG's in the media/clips folder before I submit to Frigate+ via the UI. That way I keep the bounding boxes and don't have to add them manually. I also maintain the false positives as well.

**Reply by jakeycrx** [reply]
I can report that masking doesn't appear to cause any issues. I was at 718 images with no masking and everything annotated with model 2024.1. I then added 364 masked images with 2024.2 (all cats/dogs) I decided to go with grey as it's more neutral than black. It probably makes no difference, but it didn't do any harm. There are a lot of cars in my street, so I add quite a lot of masking, especially on the camera I have at the front of the house which is mounted to the soffit.

![front-1725690836 993692-1qgfnn-clean](https://github.com/user-attachments/assets/9cf0b645-2b5b-4a49-887d-fffd79eb4441)
![reolink-1725853903 896071-yuctow-clean](https://github.com/user-attachments/assets/0950b9dd-b13b-4f96-8774-517fb55a0935)
![dahua-1725555903 583431-17rir6-clean](https://github.com/user-attachments/assets/082a58fd-7ff8-44c6-88c4-859da12ade11)

My main aim in all of this is to correctly differentiate between cats and dogs. I have an outside speaker which plays the sound of angry cat when a cat is detected, which is very effective at stopping cats urinating and defecating everywhere. People walking their dogs look a bit shocked when their dog is incorrectly detected as a cat though. 

I do have a condition not to trigger the cat scarer if a dog has been detected within the last 2 minutes, and to kill the cat scarer if a dog is detected while playing, but certain dogs really do look like a cat in the dark.

It's still occasionally getting a cat/dog confused, but that's mainly because the cat/dog is far away from the camera, it's dark, or it's looking down on my dog. I could reduce the zones a bit to deal with the distance issue, but it's quite amusing watching the cats run away when the cat scarer is triggered. With some more training, I think it will be almost 100%. 

These are some snapshots after I did the training with masking:

![image](https://github.com/user-attachments/assets/91e7c940-47be-4523-80d9-1665c36b2f0d)
![image](https://github.com/user-attachments/assets/93d41212-07a0-4957-935b-adc33d1c68fc)
![image](https://github.com/user-attachments/assets/a69c02b8-3a4f-452f-abf2-af4871576e2b)

This is my dog and as you can see, she's very cat like:

![image](https://github.com/user-attachments/assets/e347b83e-5f8c-41f7-ad81-60bfc9b95944)
![image](https://github.com/user-attachments/assets/b8ba5b38-46d3-4367-bdf0-6b6aad731461)

I did a lot of training on images of my dog, and it detects her a lot better now.  It does still detect her as a cat though. I will do some more training on her.

![image](https://github.com/user-attachments/assets/3dc3ab5f-adce-4153-a124-0cde61c12c4b)
![image](https://github.com/user-attachments/assets/36b5c05e-9c11-41ba-9ae4-a0dc12d562c0)

A few questions:

- Would you consider adding masking to the Frigate+ UI?
- Are you still planning to add a feature to scroll through images in the Frigate+ UI? With zooming now available, I can see a lot of my bounding boxes need tightening up, but it's very difficult to check all the images with the current setup.
- What happens if we run out of trainings? I don't mind purchasing more.

**Reply by blakeblackshear** [reply]
Here is the roadmap: https://github.com/orgs/frigate-nvr/projects/2

> Would you consider adding masking to the Frigate+ UI

I have it listed on the annotator improvements already. I'm not sure how necessary it will be after suggested labels are added. I definitely don't want the images masked out like this directly in the image data. I wouldn't want to pull them into the base model, so I will need to think through the right approach.

> Are you still planning to add a feature to scroll through images in the Frigate+ UI? With zooming now available, I can see a lot of my bounding boxes need tightening up, but it's very difficult to check all the images with the current setup.

Yes. I have already laid the groundwork for this so its possible to go back through images when you add a new label to a camera. If you add a new label to a camera right now, then all that camera's images will lose their green check.

> What happens if we run out of trainings? I don't mind purchasing more.

I still need to add the ability to purchase more in 5 packs. There haven't been that many users who have run out, so I have prioritized other things.

**Reply by shinyzard123** [reply]
Do you have an ETA on openvino support being finished? I am looking to setup another system at a different location and would prefer not to buy a coral if it's not needed

**Reply by bagobones** [reply]
I would like the option to submit only a cropped object sometimes.. I have indoor and outdoor cameras and many times I don't submit due to personally identifiable info or family in the photos, but maybe I would like to submit a bunch of weak or false dog / cat detections.

**Reply by ivanjx** [reply]
@shinyzard123 in the email, blake told me around 4-6 weeks. should be the same time the next base model version release.

**Reply by jakeycrx** [reply]
> I have it listed on the annotator improvements already. I'm not sure how necessary it will be after suggested labels are added. I definitely don't want the images masked out like this directly in the image data. I wouldn't want to pull them into the base model, so I will need to think through the right approach.

Suggested labels should help. This gives you an idea of what I'm working with regarding cars.

![dahua-1726852429 301819-bjx9l7-clean](https://github.com/user-attachments/assets/b173bceb-113a-4c73-8072-da70f1892763).

> Yes. I have already laid the groundwork for this so its possible to go back through images when you add a new label to a camera. If you add a new label to a camera right now, then all that camera's images will lose their green check.

It would be good to have the option to scroll through images at any time to check labels and bounding boxes. 

> I still need to add the ability to purchase more in 5 packs. There haven't been that many users who have run out, so I have prioritized other things.

I'd be happy to purchase a pack of 5.

**Comment by Anc0dia** [comment]
I noticed you are adding new object labels for the 2024.3 release.   Not sure how to request for an object label to be added, but I'd like to add skunk to the list. Thanks.
